libcxx/include/__expected/expected.h

```diff
@@ -136,14 +134,7 @@ class expected {
     noexcept(is_nothrow_copy_constructible_v<_Tp> && is_nothrow_copy_constructible_v<_Err>) // strengthened
     requires(is_copy_constructible_v<_Tp> && is_copy_constructible_v<_Err> &&
              !(is_trivially_copy_constructible_v<_Tp> && is_trivially_copy_constructible_v<_Err>))
-      : __has_val_(__other.__has_val_) {
```

Comment:
I think we need a test for each of these changed functions. It should be really easy to add. For example, if we have

```
std::expected<TailClobberer<0>, bool> e = {...};
std::expected<TailClobberer<0>, bool> copy(e);
assert(copy.has_value()); // would be false before the patch, should be true after
```

Reply:
Agreed, I'm on it :)

Reply:
Alright, I added a bunch of tests that should cover all the modified functions.